main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -191,6 +200,39 @@ object BuildServerProtocol {
     )
   }
 
+  private def checkMetalsCompatibility(
+      semanticdbEnabled: Boolean,
+      semanticdbVersion: String,
+      params: InitializeBuildParams,
+      log: Logger
+  ): Unit = {
+    for {
+      data <- params.data
+      // try parse metadata as MetalsMetadata
+      metalsMetadata <- Converter.fromJson[MetalsMetadata](data).toOption
+    } {
+      if (!semanticdbEnabled) {
+        log.warn(s"${params.displayName} requires the semanticdb compiler plugin")
+        log.warn(
+          s"Please consider setting 'Global / semanticdbEnabled := true' in your global sbt settings ($$HOME/.sbt/1.0)"
```

Comment:
As a style issue, we should drop "Please" and also start with lower case:

```suggestion
          s"consider setting 'Global / semanticdbEnabled := true' in your global sbt settings ($$HOME/.sbt/1.0)"
```